sklearn/cluster/_kmeans.py

```diff
@@ -308,10 +308,18 @@ def k_means(
         - If a callable is passed, it should take arguments `X`, `n_clusters` and a
           random state and return an initialization.
 
-    n_init : int, default=10
+    n_init : 'auto' or int, default=10
         Number of time the k-means algorithm will be run with different
         centroid seeds. The final results will be the best output of
-        `n_init` consecutive runs in terms of inertia.
+        n_init consecutive runs in terms of inertia.
+
+        When `n_init='auto'`, the number of runs will be 10 if using
+        `init='random'`, and 1 if using `init='kmeans++'`.
+
+        .. versionadded:: 1.2
+           Added 'auto' option for `n_init`.
+        .. deprecated:: 1.2
+           Default value for `n_init` will change from 10 to `'auto'` in 1.4
```

Comment:
I think we will end up with following https://github.com/scikit-learn/scikit-learn/pull/23244, which means:

```suggestion
        .. versionchanged:: 1.4
           Default value for `n_init` will change from 10 to `'auto'` in version 1.4
```